File: Level1/완주하지 못한 선수.py
def solution(participant, completion):

    runners = {}
    for c in completion:
        if c not in runners:
            runners[c] = 1
        else:
            runners[c] += 1
    for p in participant:
        if p not in runners:
            return p
        elif runners[p] == 0:
            return p
        else:
            runners[p] -= 1

File: Level1/완주하지 못한 선수.java
import java.util.HashMap;

class Solution {
    public String solution(String[] participant, String[] completion) {
        String answer = "";
        HashMap<String, Integer> hashMap = new HashMap<>();
        for (String c: completion) {
            if (hashMap.get(c) == null) {
                hashMap.put(c, 1);
            }
            else {
                int i = hashMap.get(c) + 1;
                hashMap.put(c, i);
            }
        }

        for (String p: participant) {
            if (hashMap.get(p) == null ) {
                answer = p;
                break;
            }
            if (hashMap.get(p) == 0) {
                answer = p;
                break;
            }
            else {
                int i = hashMap.get(p) - 1;
                hashMap.put(p, i);
            }
        }
        return answer;
    }
}